IddEnums_constexpr.hpp

```diff
@@ -0,0 +1,7232 @@
+     * match a name or a description (case insensitive). */
+  constexpr EnumBase(std::string_view t_value) : m_value(lookupValue(t_value)) {}
+
+  /** Returns the name associated with t_value, if it exists. Otherwise,
+     * throws std::runtime_error. */
+  static constexpr std::string_view valueName(int t_value) {
+    t_value = lookupValue(t_value);
+    return Enum::names[static_cast<size_t>(t_value)];
+  }
+
+  /** Returns the description associated with t_value, if it exists.
+     * Otherwise, throws std::runtime_error. */
+  static constexpr std::string_view valueDescription(int t_value) {
+    t_value = lookupValue(t_value);
+    auto& desc = Enum::descriptions[static_cast<size_t>(t_value)];
+    return desc.empty() ? valueName(t_value) : desc;
+  }
+
+  /** Returns the set of all values in this enum's domain. */
+  static constexpr auto getValues() {
+    std::array<int, Enum::Size> values;
+    std::iota(values.begin(), values.end(), typename Enum::SizeType(0));
+    return values;
+  }
+
+  /** Returns this instance's current value (as an integer). */
+  constexpr int value() const {
+    return m_value;
+  }
+
+  /** Returns the name associated with this instance's current value. */
+  constexpr std::string_view valueName() const {
+    return valueName(m_value);
+  }
+
+  /** Returns the description associated with this instance's current value.
+     */
+  constexpr std::string_view valueDescription() const {
+    return valueDescription(m_value);
+  }
+
+  /** Set this instance's value to t_value. Throws std::runtime_error if
+     *  t_value is not in the domain. */
+  constexpr void setValue(int t_value) {
+    m_value = lookupValue(t_value);
+  }
+
+#if 1
+  constexpr auto operator<=>(const EnumBase<Enum>&) const = default;
+#else
+  constexpr bool operator==(const EnumBase<Enum>& other) const {
+    return (m_value == other.m_value);
+  }
+  constexpr bool operator!=(const EnumBase<Enum>& other) const {
+    return (m_value != other.m_value);
+  }
+  constexpr bool operator>(const EnumBase<Enum>& other) const {
+    return (m_value > other.m_value);
+  }
+  constexpr bool operator>=(const EnumBase<Enum>& other) const {
+    return (m_value >= other.m_value);
+  }
+  constexpr bool operator<(const EnumBase<Enum>& other) const {
+    return (m_value < other.m_value);
+  }
+  constexpr bool operator<=(const EnumBase<Enum>& other) const {
+    return (m_value <= other.m_value);
+  }
+#endif
+
+  /** Returns the (integer) value associated with t_name, as determined by
+     * case-insensitive comparison to the enumerated names and descriptions. */
+  constexpr static int lookupValue(std::string_view t_name) {
+    // auto nameUC = ascii_to_upper_copy(t_name);  // TODO: make this constexpr
+
+    std::array<char, Enum::max_element_size> arr{};  // vector<char> works on recent clang, not on GCC 10
+    std::transform(t_name.begin(), t_name.end(), arr.begin(), charToUpper);
+    std::string_view nameUC{arr.data()};
```

Comment:
Uppercasing a string_view via a std::array<char> in between... Enum::max_element_size is dynamically calculated at compile time, see next